hal/src/rtl872x/ble_hal.cpp

```diff
@@ -1626,7 +1628,12 @@ int BleGap::startScanning(hal_ble_on_scan_result_cb_t callback, void* context) {
                 }
                 HAL_Delay_Milliseconds(10);
             }
-            SPARK_ASSERT(!isScanning_);
+            if (isScanning_) {
+                stop();
```

Comment:
What if it's deadlocked etc? Should we check return values here and SPARK_ASSERT if it fails to reinitialize perhaps?

Reply:
sounds good.